### PR TITLE
ci: add e2e tests

### PR DESCRIPTION
**Pull request by jgrer**
- Added new workflow e2e.yaml, which runs on new PR's and pushes to the main branch
- Designed workflow to install required node packages and docker desktop, then run jest tests
- Added new folder e2e, which contains the yarn packages needed to run the tests and setup for typescript
- Created test file "test-copa-extension.test.ts"
- Added test which inputs "nginx:1.21.6", runs a scan, and confirms that the patched image was created

**Closes**: #6

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Docker Inc.
```
Comment (sozercan) [inline .github/workflows/e2e.yaml:1]: do we need the copyright and license text?

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - main
+  pull_request:
+
+jobs:
+  e2e:
+    name: Run E2E tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (sozercan) [inline .github/workflows/e2e.yaml:29]: please add the digest instead of tags. see existing actions for examples

File: .github/workflows/e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+      - name: Run E2E tests
+        env:
+          SKIP_EXTENSION_IMAGE_BUILD: true
+        run: |
+          cd e2e
+          yarn install
+          xvfb-run --auto-servernum --server-args="-screen 0 1920x1080x24" -- yarn test:e2e
```
Comment (sozercan) [inline .github/workflows/e2e.yaml:58]: nit: add new lines. applies below too

File: e2e/e2e.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Docker Inc.
```
Comment (sozercan) [inline e2e/e2e.yaml:1]: why is this duplicated?

**Comment by ashnamehrotra** [comment]
could we try downgrading the dependency review due to https://github.com/actions/dependency-review-action/issues/786?

File: e2e/e2e.yaml (hunk 4 above)
Comment (jgrer) [reply e2e/e2e.yaml]: not sure, removed.

File: e2e/.eslintrc [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  ],
+  "rules": {
+    "prettier/prettier": [
+      "error"
+    ]
+  },
+  "globals": {
+    "page": true
+  }
+}
```
Comment (ashnamehrotra) [inline e2e/.eslintrc:24]: nit: newline

File: e2e/.gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+screenshots
+.yarn
+node_modules
```
Comment (ashnamehrotra) [inline e2e/.gitignore:3]: nit: newline

**Comment by jgrer** [comment]
> could we try downgrading the dependency review due to [actions/dependency-review-action#786](https://github.com/actions/dependency-review-action/issues/786)?

doesn't seem to work, will wait for update to the action

File: .github/workflows/dependency-review.yml [hunk 7]
```diff
@@ -24,4 +24,4 @@ jobs:
       - name: 'Checkout Repository'
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       - name: 'Dependency Review'
-        uses: actions/dependency-review-action@5a2ce3f5b92ee19cbb1541a4984c76d921601d7c # v4.3.4
+        uses: actions/dependency-review-action@0c155c5e8556a497adf53f2c18edabf945ed8e70 # v4.3.2
```
Comment (ashnamehrotra) [inline .github/workflows/dependency-review.yml:27]: we can revert this back to latest while they fix the action

**Review by ashnamehrotra — APPROVED** [review]
LGTM

**Pull request merged**